src/AccessibilityInsights.SharedUx/Controls/TestTabs/ColorContrast.xaml.cs

```diff
@@ -111,21 +112,49 @@ public void SetElement(ElementContext ec)
             }
         }
 
+        private static AnalyzerVersion SelectAnalyzerVersion(Bitmap bitmap)
+        {
+            int versionThresholdSize = ConfigurationManager.GetDefaultInstance().AppConfig.ColorContrastVersionThresholdSize;
+            return (bitmap.Width * bitmap.Height) > versionThresholdSize ? AnalyzerVersion.V1 : AnalyzerVersion.V2;
+        }
+
         private void RunAutoCCA(Bitmap bitmap)
         {
-            var bmc = new BitmapCollection(bitmap, new ColorContrastConfigBuilder().Build());
-            var result = bmc.RunColorContrastCalculation();
-            var pair = result.MostLikelyColorPair;
+            AnalyzerVersion analyzerVersion = SelectAnalyzerVersion(bitmap);
+            Stopwatch stopwatch = Stopwatch.StartNew();
+            IColorContrastResult result = null;
 
-            if (pair == null)
+            try
             {
-                throw new InvalidOperationException("Unable to determine colors!");
-            }
+                var bmc = new BitmapCollection(bitmap,
+                    new ColorContrastConfigBuilder().WithAnalyzerVersion(analyzerVersion).Build());
+                result = bmc.RunColorContrastCalculation();
+                var pair = result.MostLikelyColorPair;
+
+                stopwatch.Stop();
 
-            SetConfidenceVisibility(Visibility.Visible);
-            this.ContrastVM.FirstColor = pair.DarkerColor.DrawingColor.ToMediaColor();
-            this.ContrastVM.SecondColor = pair.LighterColor.DrawingColor.ToMediaColor();
-            tbConfidence.Text = result.Confidence.ToString();
+                if (pair == null)
+                {
+                    throw new InvalidOperationException("Unable to determine colors!");
+                }
+
+                SetConfidenceVisibility(Visibility.Visible);
+                this.ContrastVM.FirstColor = pair.DarkerColor.DrawingColor.ToMediaColor();
```

Comment:
Could we:

```suggestion
                ContrastVM.FirstColor = pair.DarkerColor.DrawingColor.ToMediaColor();
```

Reply:
Short answer: I'd rather not do this as a part of this PR.

Longer answer: We should talk more about this in the weeks to come. The original authors of this code were from a team where the coding standard was to _always_ explicitly include the `this.` prefix, so that's how most of the code base ha been written. Those people are no longer on the team, and much of our newer code omits the `this.` prefix. I'm totally fine with removing the prefixes from the code, but I'd argue that we ought to approach it as a series of PR's that change _just that style_ as a group, rather than gradually changing it one PR at a time. 

The .editorconfig file has [4 lines](https://github.com/microsoft/accessibility-insights-windows/blob/main/src/.editorconfig#L29-L33) that we might be able to use to leverage as we work through these. It's easy work, but it takes time. The trick is to find a balance between what we'd like to do in the code compared to what we need to accomplish across the full suite of Accessibility Insights products.